### PR TITLE
Use stream_limiter again using its new, fixed version

### PR DESCRIPTION
**Pull request by litchipi**
Fixing errors when dealing with TCP Stream in stream_limiter

Right now it works fine and parametric tests can attest this

Before merging this PR, let's make sure everything is 100% OK when using this branch of Peernet on realworld scale example.
I found that the default are a bit too low for some use cases, so maybe raise them a bit.

Also, the code in this PR aims at the git repository of `stream_limiter`, let's create a release and point to it first

### CODE REVIEW

**Comment by litchipi** [comment]
When using `stream_limiter`, the labnet should still  be able to reach 3000 txps without having orange blocks.
Adapt the configuration in consequence and test this so we can validate the working setup

**Comment by AurelienFT** [comment]
> When using `stream_limiter`, the labnet should still be able to reach 3000 txps without having orange blocks. Adapt the configuration in consequence and test this so we can validate the working setup

In the latest main version we reached 10k txps without problems can you check that is still works for this amount of txps also ?

File: src/peer.rs [hunk 1]
```diff
@@ -159,6 +159,7 @@ pub(crate) fn new_peer<
         ) {
             Ok(peer_id) => peer_id,
             Err(_) => {
+                println!("ERROR DURING HANDSHAKE");
```
Comment (AurelienFT) [inline src/peer.rs:162]: remove this print

**Review by AurelienFT — COMMENTED** [review]
Nice job ! Ty @litchipi !!!!

File: src/transports/tcp.rs [hunk 2]
```diff
@@ -88,26 +88,30 @@ impl Default for TcpConnectionConfig {
 pub struct TcpEndpoint {
     pub config: TcpConnectionConfig,
     pub address: SocketAddr,
-    pub stream: TcpStream,
+    pub stream: Limiter<TcpStream>,
```
Comment (modship) [inline src/transports/tcp.rs:91]: maybe  call it `stream_limiter` to avoid `endpoint.stream.stream` call which is not very "elegant"

**Review by modship — APPROVED** [review]

**Review by AurelienFT — APPROVED** [review]

**Comment by litchipi** [comment]
(Waiting to test the default settings on the labnet with ~10k tps)

**Comment by litchipi** [comment]
I'm not entirely satisfied with the tests on the labnet using the stream_limiter

# Without any stream limiting

This is the labnet without any rate limiting (currently in testnet)

Capture d’écran du 2023-06-21 09-28-19

# Using 50Mo / secs limiting

When using the limiter with a limit of 50 Mo/secs, everything runs well, but after some time we start to see this

Capture d’écran du 2023-06-21 09-45-11

Capture d’écran du 2023-06-21 09-48-24

# Using `u64::MAX` as rate limit

Capture d’écran du 2023-06-21 10-36-59

**Comment by AurelienFT** [comment]
Can you fix the conflicts and tests so that we can merge it ? @litchipi

**Comment by litchipi** [comment]
Got a clean labnet now with 10000 txps, no block misses at all.

For debugging purposes, I added the `log` crate and putted some `log::error` when a message fails to send, should I remove it ? I think it can be kept

**Comment by litchipi** [comment]
The new `send_timeout` test doesn't work, because the `write_exact_timeout` functions are broken with the stream limiter.
I'll try to rework the functions so they pass and time out correctly

**Comment by litchipi** [comment]
@AurelienFT @modship Ready for a final review before merge, the CI should be all green now

File: src/transports/mod.rs [hunk 3]
```diff
@@ -174,6 +174,7 @@ impl<Id: PeerId> Transport<Id> for InternalTransportType<Id> {
         data: &[u8],
         timeout: Duration,
     ) -> PeerNetResult<()> {
+        println!("Transports send timeout");
```
Comment (AurelienFT) [inline src/transports/mod.rs:177]: Remove this print

File: src/transports/quic.rs [hunk 4]
```diff
@@ -654,6 +654,7 @@ impl<Id: PeerId> Transport<Id> for QuicTransport<Id> {
         data: &[u8],
         timeout: Duration,
     ) -> PeerNetResult<()> {
+        println!("Quic send timeout");
```
Comment (AurelienFT) [inline src/transports/quic.rs:657]: Remove this print

File: src/network_manager.rs [hunk 5]
```diff
@@ -142,7 +142,6 @@ impl<Id: PeerId> ActiveConnections<Id> {
     }
 
     pub fn remove_connection(&mut self, id: &Id) {
-        println!("Removing connection from: {:?}", id);
```
Comment (AurelienFT) [inline src/network_manager.rs:145]: Don't remove this it's useful to debug the logs.

**Review by AurelienFT — APPROVED** [review]

**Comment by AurelienFT** [comment]
I'm merging it to be up to date for new PRs. Good job @litchipi sorry to remove you the pleasure to click this green button

**Pull request merged**